JSON.cpp

```diff
@@ -69,9 +69,30 @@ JSON JSON::parseFromString(std::string String)
 				x++;
 			}
 			String.erase(0, x + 1);
-
-
-
+		}
+		else if (akt1 == "monsters") {
+			size_t x = String.find('[') + 1;
+			while ((String[x] != ']') && (x != String.size())) {
+				if (String.find('"') != std::string::npos) {
+					x = String.find('"') + 1;
+				}
+				else { x = String.size(); }
+				while ((String[x] != '"') && (x != String.size())) {
+					if (isupper(String[x]) && (String[x - 1] != '-') && (String[x - 1] != '_') && (akt2 != "")) {
+						akt2 = akt2 + ' ' + String[x];
+					}
+					else if (String[x] != ' ') {
+						if (String[x] == 'o' && String[x + 1] == 'f') {
+							akt2 = akt2 + ' ' + String[x];
+						}
```

Comment:
Ezt a részt nem igazán értem. Ha "of" van, akkor elé kerüljön egy szóköz, és úgy adja hozzá az 'o'-t? Az else ág alapján ha pl. "oc" vagy "nf" van, akkor szóköz nélkül adja hozzá az 'o'-t vagy 'n'-t? Más parsereit nehéz átlátni kommentek nélkül, de nekem ez tényleg fura. Amúgy ettől még approve-olom, mert késve review-olok.

Reply:
Tudom, hogy rossz PR-hoz írtam, de nem onnan lett ágaztatva, amihez megnyitva, ezért mindenhol jelzi változásnak.

Reply:
Azért volt erre szükség, hogy "Prince Aidan of Khanduras" nevében az "of" szót ne nagy "O"-val olvassuk be, mert minden szóköz után a következőnek beolvasott string-et nagybetűvel olvassuk be, mint nevet, de ugye ez az "of" csak egy kötőszócska, és így a unittestben hibára futott a név ellenőrzésünk. Lehet, hogy van erre szebb és jobb megoldás, nekünk ez jutott eszünkbe. Köszi szépen az approve-ot! 😄

Reply:
Hát egy ellenőrzést betennék elé, hogy x legyen kisebb, mint string-hossz mínusz 1

Reply:
Pontosan hogyan is érted ezt az ellenőrzést? Mert ezt elvileg már lekezeltük a kódunkban.